xbmc/addons/Addon.cpp

```diff
@@ -622,6 +621,52 @@ AddonVersion CAddon::GetDependencyVersion(const std::string &dependencyID) const
   return AddonVersion("0.0.0");
 }
 
+
+void OnPreInstall(const AddonPtr& metaAddon)
```

Comment:
Because it's not the actual local addon, but a description of an addon created from the repo which contain info you cannot trust to be the same as locally, nor exist locally. Perhaps the name wasn't as clear as i thought..